nav2_planner/src/planner_server.cpp

```diff
@@ -705,7 +705,7 @@ void PlannerServer::isPathValid(
       {
         response->is_valid = false;
         break;
-      } else if (cost == nav2_costmap_2d::LETHAL_OBSTACLE) {
+      } else if (cost == nav2_costmap_2d::LETHAL_OBSTACLE || cost > request->max_cost) {
```

Comment:
I think it should be in the `use_radius` condition as well, this seems related to both SE2 footprint and radius checks